consensus/bitcoin-ng.md

```diff
@@ -84,9 +84,41 @@ Bitcoin-NG inherits Bitcoin's ability to fork in case of multiple block producer
         mempool content MB(1)' and MB(2) should be very similar. Re-signing
         the delta of transactions missing in Leader 2's node is very cheap operation.
 
+Proof of Fraud
+==
 
-
-References:
+A malicious leader could construct forks in a generation of micro
+blocks, either by forking directly from his key block, or by forking
+on a micro block. This could be done to disrupt the network or to
+perform double spend attacks.
+
+To prevent this malicious behaviour, the miner of the next key block
+can submit Proof-of-Fraud in a micro block in its generation. Only
+this leader can submit Proof-of-Fraud to avoid flooding the network
+with reports, which in itself could be viewed as a DoS
+attack. Furthermore, reporting a key block more than once is regarded
+as a protocol violation, even if there is more than one act of fraud
+committed.
+
+The Proof-of-Fraud is constructed of two sibling micro headers (with
+the same previous pointer) along with the public key of the
+signer. The Proof-of-Fraud is included in a micro block body, and the
+header must contain the hash of the Proof-of-Fraud object to mark its
+existence in the block.
+
+If a miner is reported as fraudulent in this manner, its beneficiary
+will not get any mining reward for that key block (i.e., no coinbase
+and no fees for either generation). The reporter of the fraud will get
+a reward controlled by governance (initially set to 5% of the coinbase
```

Comment:
Shall the 5% be not the 5% of the coinbase only, but the 5% of the full compensation being invalidated?


From https://www.usenix.org/system/files/conference/nsdi16/nsdi16-paper-eyal.pdf

> The poison transaction has to be placed on the blockchain within the maturity window of the misbehaving leader’s key block, and before the revenue is spent by the malicious leader. Besides invalidating the compensation sent to the leader that generated the fork, a poison transaction grants the current leader a fraction of that compensation, e.g., 5%. The choice of this value is explained in Section 5.

The whitepaper does not seem to actually explain this 5% value in section 5. Though I understand that the 5% shall be 5% of "the compensation sent to the leader that generated the fork" that is getting invalidated ("Besides invalidating the compensation ... a fraction of *that* compensation ..."). Linking the fraud report reward only to the coinbase would link the economic incentive of the PoF reported only to the inflation - if any.

Reply:
Valid point. This needs to be revisited. I suggest accepting this as it stands, since it reflects the implementation. We can have a follow up ticket where this is changed if we come to the conclusion that the 5% should indeed be of the total reward. Both for code and documentation.

Reply:
Same as https://github.com/aeternity/epoch/pull/1557#discussion_r218406880
Ok.